pkg/querier/batch/stream.go

```diff
@@ -89,33 +73,74 @@ func mergeBatches(batches batchStream, result batchStream, size int) batchStream
 }
 
 func mergeStreams(left, right batchStream, result batchStream, size int) batchStream {
-	result.reset()
-	result = result[:0]
+	if cap(result) >= len(left)+len(right) {
```

Comment:
I would appreciate a comment here like "ensure that `result` has enough space to hold left and right added together".